src/racer/cargo.rs

```diff
@@ -673,9 +697,18 @@ pub fn get_crate_file(kratename: &str, from_path: &Path) -> Option<PathBuf> {
 
     if let Some(tomlfile) = find_cargo_tomlfile(from_path.to_path_buf()) {
         // look in the lockfile first, if there is one
+        // also search workspaces for a lockfile
         trace!("get_crate_file tomlfile is {:?}", tomlfile);
+
+        let absolute_path = env::current_dir().unwrap().join(from_path);
+        let workspace = find_workspace_root(absolute_path);
+
         let mut lockfile = tomlfile.clone();
         lockfile.pop();
+        if workspace.is_some() {
```

Comment:
This could use the `if let Some(workspace) = workspace` construct to avoid the subsequent `unwrap`.

Reply:
Sure thing, I'll tweak this when I'm home from work.